src/parser/smt2/smt2_lexer_new.cpp

```diff
@@ -0,0 +1,385 @@
+  }
+  // ~!@$%^&*_-+=<>.?/
+  d_symcTable[static_cast<size_t>('~')] = CharacterClass::SYMBOL;
+  d_symcTable[static_cast<size_t>('!')] = CharacterClass::SYMBOL;
+  d_symcTable[static_cast<size_t>('@')] = CharacterClass::SYMBOL;
+  d_symcTable[static_cast<size_t>('$')] = CharacterClass::SYMBOL;
+  d_symcTable[static_cast<size_t>('%')] = CharacterClass::SYMBOL;
+  d_symcTable[static_cast<size_t>('^')] = CharacterClass::SYMBOL;
+  d_symcTable[static_cast<size_t>('&')] = CharacterClass::SYMBOL;
+  d_symcTable[static_cast<size_t>('*')] = CharacterClass::SYMBOL;
+  d_symcTable[static_cast<size_t>('_')] = CharacterClass::SYMBOL;
+  d_symcTable[static_cast<size_t>('-')] = CharacterClass::SYMBOL;
+  d_symcTable[static_cast<size_t>('+')] = CharacterClass::SYMBOL;
+  d_symcTable[static_cast<size_t>('=')] = CharacterClass::SYMBOL;
+  d_symcTable[static_cast<size_t>('<')] = CharacterClass::SYMBOL;
+  d_symcTable[static_cast<size_t>('>')] = CharacterClass::SYMBOL;
+  d_symcTable[static_cast<size_t>('.')] = CharacterClass::SYMBOL;
+  d_symcTable[static_cast<size_t>('?')] = CharacterClass::SYMBOL;
+  d_symcTable[static_cast<size_t>('/')] = CharacterClass::SYMBOL;
+  d_symcTable[static_cast<size_t>(',')] = CharacterClass::SYMBOL;
+  // whitespace
+  d_symcTable[static_cast<size_t>(' ')] = CharacterClass::WHITESPACE;
+  d_symcTable[static_cast<size_t>('\t')] = CharacterClass::WHITESPACE;
+  d_symcTable[static_cast<size_t>('\r')] = CharacterClass::WHITESPACE;
+  d_symcTable[static_cast<size_t>('\n')] = CharacterClass::WHITESPACE;
+}
+
+const char* Smt2LexerNew::tokenStr() const
+{
+  Assert(!d_token.empty() && d_token.back() == 0);
+  return d_token.data();
+}
+bool Smt2LexerNew::isStrict() const { return d_isStrict; }
+bool Smt2LexerNew::isSygus() const { return d_isSygus; }
+
+Token Smt2LexerNew::nextTokenInternal()
+{
+  Trace("lexer-debug") << "Call nextToken" << std::endl;
+  d_token.clear();
+  Token ret = computeNextToken();
+  // null terminate?
+  d_token.push_back(0);
+  Trace("lexer-debug") << "Return nextToken " << ret << " / " << tokenStr()
+                       << std::endl;
+  return ret;
+}
+
+Token Smt2LexerNew::computeNextToken()
+{
+  bumpSpan();
+  char ch;
+  // skip whitespace and comments
+  for (;;)
+  {
+    do
+    {
+      if ((ch = nextChar()) == EOF)
+      {
+        return Token::EOF_TOK;
+      }
+    } while (isCharacterClass(ch, CharacterClass::WHITESPACE));
+
+    if (ch != ';')
+    {
+      break;
+    }
+    while ((ch = nextChar()) != '\n')
+    {
+      if (ch == EOF)
+      {
+        return Token::EOF_TOK;
+      }
+    }
+  }
+  bumpSpan();
+  pushToToken(ch);
+  switch (ch)
+  {
+    case '!': return Token::ATTRIBUTE_TOK;
+    case '(': return Token::LPAREN_TOK;
+    case ')': return Token::RPAREN_TOK;
+    case '|':
+      do
+      {
+        ch = nextChar();
+        if (ch == EOF)
+        {
+          return Token::UNTERMINATED_QUOTED_SYMBOL;
```

Comment:
Why is this its own token kind and not an error?

Reply:
This is to support interactive mode.  A token of this form instructs the parser to throw and EOF exception, which is caught in the interactive shell and it proceeds to read the next line.
See:
https://github.com/cvc5/cvc5/blob/main/src/parser/smt2/smt2_term_parser.cpp#L228
https://github.com/cvc5/cvc5/blob/main/src/main/interactive_shell.cpp#L381